### PR TITLE
change email address

### PR DESCRIPTION
**Pull request by xi**
This pull request serves as a hub for implementing changing ones email address. This will work a lot like changing the password (#2807): In the user settings, the user can set a new email address which also requires providing the password.

What is different here is that after setting the new email address, it will remain inactive until it has been activated (similar to initial account activation). It is not completely clear yet how much code we can reuse for this activation. Another question is what happens to existing sessions on activation.
- [x] Implement backend to change email address
- [X] Implement UI to change email address
- [x] Implement activation backend
- [x] Implement activation frontend

### CODE REVIEW

**Comment by 2e2a** [comment]
@xi, @der-john: Just committed the backend part of the email activation.

File: src/adhocracy_core/adhocracy_core/messaging/__init__.py [hunk 1]
```diff
@@ -339,6 +339,27 @@ def send_password_change_mail(self, user=IUser, request: Request=None):
                        request=request,
                        )
 
+    def send_new_email_activation_mail(self, user: IUser, activation_path: str,
+                                       new_email: str, request: Request=None):
+        """Send a avtivation mail to validate the new email of a user."""
```
Comment (xi) [inline src/adhocracy_core/adhocracy_core/messaging/__init__.py]: typo: "activation"

File: src/adhocracy_core/adhocracy_core/locale/en/LC_MESSAGES/adhocracy.po [hunk 2]
```diff
@@ -168,14 +168,52 @@ msgstr ""
...
+"Have fun and be inspired!\n"
+"\n"
+"Your \"${site_name}\" Team\n"
+"\n"
+"----------------------------------------------------------\n"
+"\n"
+"Hallo ${user_name},\n"
+"\n"
+"Diese E-Mail dient der Validierung Deiner neuen Email-Adresse. Bitte\n"
+"nutze den folgenden Link um sie aktivieren.\n"
```
Comment (xi) [inline src/adhocracy_core/adhocracy_core/locale/en/LC_MESSAGES/adhocracy.po]: typo: "um sie zu aktivieren"

File: src/adhocracy_core/adhocracy_core/rest/schemas.py [hunk 3]
```diff
@@ -463,16 +465,22 @@ def validate_activation_path(node, value):
...
             autoupdated = False
-            event = ResourceSheetModified(user, IUserBasic, request.registry, {},
+            event = ResourceSheetModified(user, IUserBasic, registry, {},
                                           {}, request, autoupdated)
             registry.notify(event)  # trigger reindex activation_path index
     return validate_activation_path
 
 
+
+
```
Comment (xi) [inline src/adhocracy_core/adhocracy_core/rest/schemas.py]: Extra whitespace

**Comment by xi** [comment]
Generally this seems to work fine. I think we can leave the activation frontend as it is.

Some questions/ideas:

- It might be a good idea to let the activation link expire after some time.
- I think at one point we discussed sending an information mail to the old email address.

**Comment by 2e2a** [comment]
> Generally this seems to work fine. I think we can leave the activation frontend as it is.

We might still consider changing the text shown after the activation to better cover both cases.

> It might be a good idea to let the activation link expire after some time.

This  happens already after a week.

> I think at one point we discussed sending an information mail to the old email address.

As far as I remember we decided not to do this, to avoid sending to much to the user.

**Comment by 2e2a** [comment]
Fixed typos

**Comment by xi** [comment]
> > I think at one point we discussed sending an information mail to the old email address.
>
> As far as I remember we decided not to do this, to avoid sending to much to the user.

The way I remember it we decided not to require activation from both accounts. This is why this would only be an information mail to the old account.

**Pull request merged**